the-collab-lab/tcl-70-smart-shopping-list
# #20 Add new food icons and add an icon background for dark mode

## Type of Changes

|     | Type                       |
| --- | -------------------------- |
|  ✓  | :sparkles: New assest     |
|  ✓  | :art: UI/UX                |


## Acceptance Criteria

- [x] New food icons have been added for a richer visual experience in the app.
- [x] Icon backgrounds have been implemented to enhance visibility over a dark background.
- [x] Light and dark mode themes for the card have been developed, ensuring a cohesive user experience across different modes.

## Related Issue

Closes #68 

## Description 

### New Food Icons

We have integrated a new set of food icons, offering a wider variety of images that match the diverse grocery items our users may add to their lists.

### Icon Backgrounds

With an understanding of the challenges dark mode presents in terms of contrast, we have added backgrounds to our icons. This ensures that they remain prominent and visually appealing in both light and dark mode settings.

## Updates

### Before
<img width="1416" alt="image" src="https://github.com/the-collab-lab/tcl-70-smart-shopping-list/assets/121919044/cfd1e2b7-3db8-4aee-b30f-0084af5cb8dd">
<img width="1416" alt="image" src="https://github.com/the-collab-lab/tcl-70-smart-shopping-list/assets/121919044/0fa34b46-0f8c-49f1-b325-5011b2089add">

### After
<img width="1416" alt="image" src="https://github.com/the-collab-lab/tcl-70-smart-shopping-list/assets/121919044/ca0cc84b-52e4-4fb4-8a87-64abf344468f">

<img width="1416" alt="image" src="https://github.com/the-collab-lab/tcl-70-smart-shopping-list/assets/121919044/87692115-e08a-482f-be03-7189e0eca975">

## Testing Steps / QA Criteria

* From your terminal, pull down this branch with `git pull origin ag-card-dark-light-mode` and check that branch out with `git checkout ag-card-dark-light-mode`.
* Run `npm start` to start the app.
* Toggle between light and dark mode settings to see the cards adapt in real-time.
* Verify the new icons and their backgrounds are displaying correctly on various devices and screen sizes.
* Ensure that the visual changes do not interfere with the functionality of the app.

## TODOs

- [ ] UI tests to ensure that light and dark modes are switching correctly.
- [ ] Accessibility tests to confirm that the new UI changes meet accessibility standards.
